govcd/vm_test.go

```diff
@@ -1448,28 +1118,96 @@ func (vcd *TestVCD) Test_UpdateVmSpecSection(check *C) {
 	check.Assert(err, IsNil)
 
 	vmSpecSection := vm.VM.VmSpecSection
-	osType := "sles10_64Guest"
-	vmSpecSection.OsType = osType
 	vmSpecSection.NumCpus = addrOf(4)
 	vmSpecSection.NumCoresPerSocket = addrOf(2)
 	vmSpecSection.MemoryResourceMb = &types.MemoryResourceMb{Configured: 768}
+	if vcd.client.Client.APIVCDMaxVersionIs(">=37.1") {
+		vmSpecSection.Firmware = "efi"
+	}
 
 	updatedVm, err := vm.UpdateVmSpecSection(vmSpecSection, "updateDescription")
 	check.Assert(err, IsNil)
 	check.Assert(updatedVm, NotNil)
 
 	//verify
-	check.Assert(updatedVm.VM.VmSpecSection.OsType, Equals, osType)
+	//check.Assert(updatedVm.VM.VmSpecSection.OsType, Equals, osType)
 	check.Assert(*updatedVm.VM.VmSpecSection.NumCpus, Equals, 4)
 	check.Assert(*updatedVm.VM.VmSpecSection.NumCoresPerSocket, Equals, 2)
 	check.Assert(updatedVm.VM.VmSpecSection.MemoryResourceMb.Configured, Equals, int64(768))
 	check.Assert(updatedVm.VM.Description, Equals, "updateDescription")
+	if vcd.client.Client.APIVCDMaxVersionIs(">=37.1") {
+		check.Assert(updatedVm.VM.VmSpecSection.Firmware, Equals, "efi")
+	}
 
 	// delete Vapp early to avoid env capacity issue
 	err = deleteVapp(vcd, vmName)
 	check.Assert(err, IsNil)
 }
 
+func (vcd *TestVCD) Test_VmBootOptions(check *C) {
+	fmt.Printf("Running: %s\n", check.TestName())
+	if vcd.client.Client.APIVCDMaxVersionIs("<37.1") {
+		check.Skip("Extended boot options and firmware choice were introduced in 37.1")
+	}
+
+	vmName := check.TestName()
+	org, err := vcd.client.GetAdminOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+	vdc, err := org.GetVDCByName(vcd.config.VCD.Vdc, false)
+	check.Assert(err, IsNil)
+	check.Assert(vdc, NotNil)
+
+	_, vm := createNsxtVAppAndVmWithEfiSupport(vcd, check)
+	check.Assert(err, IsNil)
+
+	hardwareVersion, err := vdc.GetHardwareVersion(vm.VM.VmSpecSection.HardwareVersion.Value)
+	check.Assert(err, IsNil)
+	check.Assert(hardwareVersion, NotNil)
+	fmt.Println(hardwareVersion)
```

Comment:
This print has little value without context information.
```
Running: TestVCD.Test_VmBootOptions
&{[0x14000169680 0x14000169e00 0x14000176540 0x140001768a0 0x14000176ae0 0x14000176cc0 0x14000176ea0] {    } 128 128 6275072 10 vmx-13 [1 2 4 8 16 32 64 128 256 512 1024 2048 4080 6128] [1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128] 0x140001e0200 0x140002bb288 0x140002bb268 0x140002bb24f}
```

Reply:
Thanks, removed.